Documents/AnalyticsToolDesignDocument.md

```diff
@@ -0,0 +1,48 @@
+## Problem
+
+Over the lifetime of an enlistment, sometimes a user's working directory fills up with too many entries. Generally this means having too many placeholders on their disk (have been read and pulled from remote) or too many modified files. This causes unacceptable losses in performance and greatly hinders users as they go about their daily development cycle. Additionally, if a local enlistment has been in use for a long time, the number of loose object may eventually reach the order of several thousand in which case that also contributes to the loss in performance.
+
+## Provided Statistics
+
+### Safe to access
+
+- **Modified paths** - The number of files which have been edited and have been handed over to git
+- **Git index** - The differences between modified paths and the index tracked by git
+- **Placeholders** - A total number of the placeholders being tracked in VFS4G and a list of the placeholders in the system (Note that this contains some upper level directories and thus not all of their children are included in the count without enumeration)
+- **Troublesome programs** - Similar to what is already being logged in the heartbeat, look at which services or programs are responsible for hydrating large amounts of files in the enlistment
+
+### Potentially detrimental to access on a mounted enlistment
+
+- **Placeholder details** - Crawling the file system to enumerate directories of placeholders may unintentionally further hydrate files (IE downloading file sizes), but doing so would allow for greater details such as the percent hydration of folders, the total number of hydrated files, the type of files which are most hydrated, etc.
+- **File system** - Provide enumerated details for files in the placeholder list or the modified path list
+
+## Background Information
+
+- **Modified paths** is the list of files which VFS4G has completely pulled from the remote and whose contents are on disk in their entirety. These files are handed to git via a hook when any git command requiring the list of changed files is called
+- **Placeholders** are files which have been opened by the local system with a read tag but without a write tag. They are objects tracked by VFS4G since git should not have to internally scan them since it is guaranteed that they have not been modified and accordingly do not have to be rehashed for differences. Their contents are on disk, but control over them has not been entirely ceded over to the system. They decrease performance because VFS4G has to track all of them individually, and this can cause certain `gvfs` commands to be slow
+- The **Git index** is gits internal list of files which have been modified (tracked or untracked) which is what is used for commit commands to keep score of everything going on inside of the enlistment. It is useful to this project by means of comparing it to the list of **hydrated files** to deduce if those files should truly be hydrated or if it may have occurred incidentally.

```

Comment:
`gits` -> `git's`

The index is only the files that are being tracked by git.  git untracked files are not something that are in the index, so I would have this say something like
```
is git's internal list of files which it is tracking for creating  the next `commit`.  
```

I'm also not sure what you are going to use the index for and what comparisons you will be doing.  You may want to talk a little about the skip-worktree bit and how that will be used in this comparison since those are the files that should not be in the modified paths list but could be a placeholder but might not be in either list.

Reply:
It's possible that the git index might not be a useful item in this tool. I was thinking along the lines of it containing the significant files which contribute to the bulk of work being done on any given git command (if my understanding serves me correctly) that it would contain useful information, especially when coupled with the other lists of files in the `.gvfs` directory. Do you think that the index won't end up being beneficial to the project?

Reply:
I think it could be useful in telling how many files git is tracking an of those how many git is controlling/keeping up to date.  See Stolee's example below.  You could have:
```
Paths tracked by Git:   3,123,456 (100%)
Paths git is updating:   1 (0%)
```